java-vtl-script/src/main/java/no/ssb/vtl/script/operations/hierarchy/SumHierarchyAccumulator.java

```diff
@@ -57,8 +57,10 @@ public VTLObject identity() {
         switch (sign) {
             case UNION:
                 return (left, right) -> {
-                    VTLNumber leftNumber = (VTLNumber) left;
-                    VTLNumber rightNumber = (VTLNumber) right;
+                    VTLNumber leftNumber =
+                            left == null || left.get() == null ? VTLNumber.of(0) : (VTLNumber) left;
```

Comment:
The fact that it can be a null reference or a VTLObject containing null reference makes me uncomfortable. I also see that the relative coverage is dropping, probably because one of the case actually never happens.

I have seen that in aggregation as well I think, so that's not on you, but I think we should flag this as a pitfall or a thing we should redesign somehow.

Reply:
I am not quire sure what you mean. If the source dataset contains null values, is it not expected that we get a VTLObject containing a null reference?